metaserver/src/main/scala/scala/meta/languageserver/Buffers.scala

```diff
@@ -38,6 +39,18 @@ class Buffers private (
     read(cwd.resolve(path))
   def read(path: AbsolutePath): String =
     Option(contents.get(path)).getOrElse(readFromDisk(path))
+
+  private val sources: JMap[AbsolutePath, Source] = new ConcurrentHashMap()
+  // Tries to parse and record it or fallback to an old source if it existed
+  def source(path: AbsolutePath): Option[Source] =
+    read(path)
+      .parse[Source]
```

Comment:
I propose we move the parsing logic to `Parser` https://github.com/scalameta/language-server/blob/dafe6bcab6e8f29cd145aba645320d92b982091d/metaserver/src/main/scala/scala/meta/languageserver/Parser.scala#L12 and remove `import scala.meta._` (it brings a lot into scope). 

In `Parser` I think we can also use a more lenient dialect than the default like scalafix does https://github.com/scalacenter/scalafix/blob/bb0edd04fa59c291e4174f3900314b5f56e37e91/scalafix-core/shared/src/main/scala/scalafix/internal/config/ScalafixConfig.scala#L74-L101 this allows us to parse as much syntax as possible.

Reply:
I'm not sure I'm following you here

Reply:
`DefaultDialect(String).parse[Source]` will parse a String with `DefaultDialect`. The dialect indicates what syntax is allowed like trailing commas, literal types, etc

Reply:
Check a0b0164e7d17a84e644a6a46fa846681c5a5a273, I'm not sure it's what you meant. I just used `ScalafixConfig.DefaultDialect` for parsing.